format/view-spec.md

```diff
@@ -111,12 +111,13 @@ The SQL representation stores the view definition as a SQL SELECT, with metadata
 |-------------|---------------------|----------------|-------------|
 | _required_  | `type`              | `string`       | Must be `sql` |
 | _required_  | `sql`               | `string`       | A SQL SELECT statement |
-| _required_  | `dialect`           | `string`       | The dialect of the `sql` SELECT statement (e.g., "trino" or "spark") |
+| _required_  | `dialect`           | `string`       | The dialect of the `sql` SELECT statement (e.g., "trino" or "spark"). Unless a version is provided only one view definition for each `type` and `dialect` should be present in a view definition.  |
```

Comment:
I don't think we want to include version information here. Isn't SQL generally compatible across versions? While you could add to the dialect, we expect SQL to be backward-compatible.

If additions are not supported, then the SQL will fail to be parsed or analyzed. We would not want to reject a dialect if it is a newer version because that would prevent using many SQL queries that are compatible with older versions.

Reply:
SQL is generally compatible across versions but not always.  For example from the Athena [engine versioning docs](https://docs.aws.amazon.com/athena/latest/ug/engine-versions.html):

> When you start using a new engine version, a small subset of queries may break due to incompatibilities. You can use workgroups to test your queries in advance of the upgrade by creating a test workgroup that uses the new engine or by test upgrading an existing workgroup. For more information, see [Testing queries in advance of an engine version upgrade](https://docs.aws.amazon.com/athena/latest/ug/engine-versions-changing.html#engine-versions-testing).

Reply:
> I thought the intent was to support multiple dialects, for the same type, but maybe the intent here is to say "type" is the unique key here? Which raises the question should dialect be required? It seems like a nice to have for compatibility reasons?

The `type` is basically how we are identifying the representation. Representations are like a union of different concrete representations and `type` selects it. Then `dialect` is specific to SQL. In another PR, we clarified that you can have multiple `sql` representations with different dialects.

I think you always want to have a dialect to give some indication of what to expect, and so you can map dialects  to different handling when using a view definition.

> I think the use-case here would be migrations for the cases where changes are breaking?

I'm skeptical we would do this automatically. SQL is not going to have very good compatibility. While we can help that out a little and should track dialect, the best solution will often be to either use a more strict representation (like a future IR one) or to replace the view with a new version that works.

Reply:
I'm OK not making this an identifier, if it isn't an identifier, do you think it is still worth tracking this?

I think there are still some unanswered questions around how multiple dialects should be handled (e.g. how are they expected to be added/validated by the writer?).